VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -47,26 +47,42 @@ void ViewItem::initializeForms()
 				  return self->nodesGetter(); }));
 }
 
+void ViewItem::insertColumn(int column)
+{
+	//Make sure we actually have enough columns
+	if (nodes_.size() <= column)
```

Comment:
A part of this code is identical with the code provided below for insertNode. Please extract the relevant bits into a separate private method: ViewItem::ensureMinimumGridSize(int cols, int rows) <-- or something similar.